### PR TITLE
Es6 import parsing

### PR DESCRIPTION
**Pull request by sheetalkamat**
TODO: Will add test cases around keyword and yield in names for this

### CODE REVIEW

File: src/compiler/parser.ts [hunk 1]
```diff
@@ -4556,6 +4573,101 @@ module ts {
...
+            // NamedImports:
+            //  { }
+            //  { ImportsList }
+            //  { ImportsList, }
+
+            // ImportsList:
+            //  ImportSpecifier
+            //  ImportsList, ImportSpecifier
+            parseExpected(SyntaxKind.OpenBraceToken);
+            namedImports.elements = parseDelimitedList(ParsingContext.ImportSpecifiers, parseImportSpecifier);
```
Comment (CyrusNajmabadi) [inline src/compiler/parser.ts]: you should only both parsing the delimited list if you actually got the open brace token.

File: src/compiler/parser.ts [hunk 2]
```diff
@@ -4615,6 +4729,12 @@ module ts {
             return isIdentifierOrKeyword() || token === SyntaxKind.StringLiteral;
         }
 
+        function nextTokenIsIdentifierOrKeywordOrStringLiteralOrAsteriskOrOpenBrace() {
```
Comment (CyrusNajmabadi) [inline src/compiler/parser.ts]: i think i'd prefer this just be called "nextTokenIsImportDeclarationStart" or something like that.

File: src/compiler/program.ts [hunk 3]
```diff
@@ -276,10 +276,10 @@ module ts {
 
         function processImportedModules(file: SourceFile, basePath: string) {
```
Comment (CyrusNajmabadi) [inline src/compiler/program.ts]: do we need to update this to be aware of es6 style imports?

Comment (sheetalkamat) [inline src/compiler/program.ts]: Yes but I did this as part of typeChecking

File: src/compiler/parser.ts (hunk 1 above)
Comment (sheetalkamat) [inline src/compiler/parser.ts]: Used ParseBracketedList to do this, which does exactly what you suggested

File: src/compiler/parser.ts [hunk 4]
```diff
@@ -4556,6 +4573,101 @@ module ts {
...
+        function parseImportStatement(fullStart: number, modifiers: ModifiersArray): ImportStatement {
+            var node = <ImportStatement>createNode(SyntaxKind.ImportStatement, fullStart);
+            setModifiers(node, modifiers);
+
+            // ImportDeclaration:
+            //  import ImportClause ModuleSpecifier ;
+            //  import ModuleSpecifier;
+            if (token !== SyntaxKind.StringLiteral) {
+                // ImportDeclaration:
+                node.importClause = parseImportClause();
```
Comment (mhegazy) [inline src/compiler/parser.ts]: i think the "From" belongs here and not in the parseImportClause

File: src/compiler/parser.ts [hunk 5]
```diff
@@ -4519,10 +4523,23 @@ module ts {
...
+                token === SyntaxKind.FromKeyword;
+        }
+
+        function parseImportDeclarationOrStatement(fullStart: number, modifiers: ModifiersArray): ImportEqualsDeclaration | ImportStatement {
             parseExpected(SyntaxKind.ImportKeyword);
+            if (token === SyntaxKind.StringLiteral ||
+                token === SyntaxKind.AsteriskToken ||
+                token === SyntaxKind.OpenBraceToken ||
+                (isIdentifier() && lookAhead(nextTokenIsCommaOrFromKeyword))) {
+                return parseImportStatement(fullStart, modifiers);
```
Comment (mhegazy) [inline src/compiler/parser.ts]: Why is it a "Statement" and not a "Declaration"?

File: src/compiler/program.ts (hunk 3 above)
Comment (mhegazy) [inline src/compiler/program.ts]: Can we add a syntax error after parsing that the syntax is not allowed. just so that we can check this incrementally.

**Pull request closed without merging**